src/util/result.h

```diff
@@ -8,16 +8,129 @@
 #include <attributes.h>
 #include <util/translation.h>
 
+#include <memory>
+#include <optional>
+#include <tuple>
 #include <variant>
+#include <vector>
 
 namespace util {
+namespace detail {
+//! Empty string list
+const std::vector<bilingual_str> EMPTY_LIST{};
 
+//! Helper to move elements from one container to another.
+template <typename T>
+void MoveElements(T& src, T& dest)
+{
+    dest.insert(dest.end(), std::make_move_iterator(src.begin()), std::make_move_iterator(src.end()));
+    src.clear();
+}
+
+//! Error information only allocated if there are errors or warnings.
```

Comment:
Maybe I'm confused, but I don't think I agree. Since the optional allocation of an `ErrorInfo` is a result of how `Info()` is implemented, not of how the `ErrorInfo` struct is defined - isn't that where it should be documented? Unless you're talking about the optional allocation of `failure` instead of `ErrorInfo`, in which case I would agree that this should be explained at this location (and you might want to add another one to info explaining the dynamic allocation of `m_info`)?